src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorVerifierInProcess.cs

```diff
@@ -22,6 +23,124 @@ public EditorVerifierInProcess(TestServices testServices)
         {
         }
 
+        public async Task CurrentLineTextAsync(
+            string expectedText,
+            bool assertCaretPosition = false,
+            bool trimWhitespace = true,
+            CancellationToken cancellationToken = default)
+        {
+            if (assertCaretPosition)
+            {
+                await CurrentLineTextAndAssertCaretPositionAsync(expectedText, trimWhitespace, cancellationToken);
+            }
+            else
+            {
+                var view = await TestServices.Editor.GetActiveTextViewAsync(cancellationToken);
+                var lineText = view.Caret.Position.BufferPosition.GetContainingLine().GetText();
+
+                if (trimWhitespace)
+                {
+                    lineText = lineText.Trim();
+                }
+
+                Assert.Equal(expectedText, lineText);
+            }
+        }
+
+        private async Task CurrentLineTextAndAssertCaretPositionAsync(
+            string expectedText,
+            bool trimWhitespace,
+            CancellationToken cancellationToken)
+        {
+            var expectedCaretIndex = expectedText.IndexOf("$$");
+            if (expectedCaretIndex < 0)
+            {
+                throw new ArgumentException("Expected caret position to be specified with $$", nameof(expectedText));
+            }
+
+            var expectedCaretMarkupEndIndex = expectedCaretIndex + "$$".Length;
+
+            var expectedTextBeforeCaret = expectedText.Substring(0, expectedCaretIndex);
```

Comment:
nit - maybe have a small helper to extract the text w/out the $$ since it looks like you do it here and on line 129

Reply:
➡️ It looks like this is cleaned up a bit in a pull request I'm submitting after this one, so I'll leave it here for now.